src/corva/runners/task.py

```diff
@@ -0,0 +1,47 @@
+from typing import Callable
+
+import requests
+
+from corva.api import Api
+from corva.models.task import TaskContext, TaskEvent, TaskStatus
+
+
+def get_task_data(api: Api, task_id: str) -> TaskEvent:
+    response = api.get(path=f'v2/tasks/{task_id}')
+    response.raise_for_status()
+
+    return TaskEvent(**response.json())
+
+
+def update_task_data(
+    api: Api, task_id: str, status: TaskStatus, data: dict
+) -> requests.Response:
+    """Updates the task. Should never raise."""
+
+    return api.put(path=f'v2/tasks/{task_id}/{status}', data=data)
+
+
+def task_runner(fn: Callable, context: TaskContext):
+    try:
+        # TODO: in case this raises, should exc be added to fail_reason?
```

Comment:
@ville what do you think about this?

Reply:
If this throws an error, then most likely update_task_data will fail as well.

Reply:
Yes. `update_task_data` doesnt raise, so the task will just silently finish without updating the task. So the task will stay in running state